addons/website_sale/controllers/main.py

```diff
@@ -359,9 +411,29 @@ def _prepare_product_values(self, product, category, search, **kwargs):
     def pricelist_change(self, pl_id, **post):
         if (pl_id.selectable or pl_id == request.env.user.partner_id.property_product_pricelist) \
                 and request.website.is_pricelist_available(pl_id.id):
+            redirect_url = request.httprequest.referrer
+            if redirect_url:
+                decoded_url = url_parse(redirect_url)
+                args = url_decode(decoded_url.query)
+                min_price = args.get('min_price')
+                max_price = args.get('max_price')
+                if min_price or max_price:
+                    try:
+                        min_price = float(min_price)
+                    except (ValueError, TypeError):
+                        pass
+                    try:
+                        max_price = float(max_price)
+                    except (ValueError, TypeError):
+                        pass
+                    previous_price_list = request.website.get_current_pricelist()
+                    previous_compute_currency = self._get_compute_currency(previous_price_list.currency_id, pl_id.currency_id)
+                    args['min_price'] = min_price and str(previous_compute_currency(min_price))
```

Comment:
This should also go inside the try/catch, or it doesn't make sense
Eg, go to `http://localhost:8069/shop?min_price=759.52&max_price=aze`
Change pricelist, it will try to parse max_price -> will crash -> it is handled by the catch, but then it will still call `_get_compute_currency` with `aze` as value